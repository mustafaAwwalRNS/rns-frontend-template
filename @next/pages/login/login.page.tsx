import React from "react";
import { GuestPanel, LoginPage } from "@components/templates";
import { useLogin } from "@queries";

/**
 * Login
 *
 * Features:
 * - Renders the login form with styling applied for better UX.
 */
export const Login = (): JSX.Element => {
  const [login, { status, loading, error }] = useLogin();
  return (
    <LoginPage
      loginFormProps={{
        loading,
        error,
        success: status === "success",
        onSubmit: login,
      }}
    />
  );
};

Login.getLayout = (page: React.ReactNode) => <GuestPanel>{page}</GuestPanel>;
