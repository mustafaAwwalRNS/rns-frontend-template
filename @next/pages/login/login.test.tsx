import React from "react";
import { render } from "test-utils";
import { Login } from "./login.page";
const setup = () => {
  const { container } = render(<Login />);
  return container;
};

describe("<Login />", () => {
  it("should match snapshot", () => {
    const login = setup();
    expect(login).toMatchSnapshot();
  });
});
