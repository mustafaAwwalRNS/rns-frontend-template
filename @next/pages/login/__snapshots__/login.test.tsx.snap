// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  color: black;
}

.c5 {
  border-top: 1px solid #ccc;
  margin: 0;
}

.c7 {
  outline: none;
  height: 48px;
  background-color: white;
  color: black;
  border: 1px solid #ccc;
  border-radius: 0.5rem;
  padding: 0.5rem;
  width: 100%;
}

.c6 {
  text-transform: uppercase;
  font-weight: bold;
  display: inline-block;
  color: black;
  margin-bottom: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: calc(-1 * 0.5rem);
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  padding: 0.5rem;
}

.c1 {
  background-color: white;
  border-radius: 0.5rem;
  padding: 2rem;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  height: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  border: none;
  cursor: pointer;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: purple;
  color: white;
  border-radius: 0.5rem;
  width: 100%;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      spacing="4"
    >
      <form
        aria-label="Login Form"
      >
        <div
          class="c2"
          role="grid"
          spacing="1"
        >
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <h1
              class="c4"
            >
              Login
            </h1>
          </div>
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <hr
              class="c5"
            />
          </div>
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <label
              class="c6"
              for="email"
            >
              email
            </label>
            <input
              class="c7"
              id="email"
              name="email"
              placeholder="Email"
              value=""
            />
          </div>
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <label
              class="c6"
              for="password"
            >
              password
            </label>
            <input
              class="c7"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <hr
              class="c5"
            />
          </div>
          <div
            class="c3"
            role="gridcell"
            spacing="1"
          >
            <button
              class="c8"
              type="submit"
            >
              Login
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
